cmd/create.go

```diff
@@ -0,0 +1,245 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	log "github.com/charmbracelet/log"
+	"github.com/spf13/cobra"
+	"github.com/thi-startup/spitfire/internal/config"
+	"github.com/thi-startup/spitfire/internal/drive"
+	"github.com/thi-startup/spitfire/internal/get"
+	"github.com/thi-startup/spitfire/utils"
+	"golang.org/x/net/context"
+)
+
+type rootfs struct {
+	size   string
+	fstype string
+	name   string
+	image  string
+	init   bool
+	path   string
+}
+
+func MakeCreateCmd() *cobra.Command {
+	command := &cobra.Command{
+		Use:          "create [flags] <image>",
+		Short:        "create a microvm",
+		Aliases:      []string{"c"},
+		SilenceUsage: true,
+		Args:         cobra.ExactArgs(1),
+	}
+
+	command.Flags().StringP("size", "s", "", "size of the rootfs")
+	command.Flags().StringP("fstype", "f", "ext4", "filesystem type. eg: ext2, ext4, btrfs, etc")
+	command.Flags().StringP("name", "n", "rootfs.ext4", "name of the loop device (defaults to rootfs.ext4).")
+	command.Flags().BoolP("init", "", false, "create a loop drive with the init flashed unto it")
+	command.Flags().StringP("image", "i", "", "docker image to create rootfs from")
+
+	command.RunE = func(cmd *cobra.Command, args []string) error {
+		name, _ := cmd.Flags().GetString("name")
+		fstype, _ := cmd.Flags().GetString("fstype")
+		size, _ := cmd.Flags().GetString("size")
+		init, _ := cmd.Flags().GetBool("init")
+		image, _ := cmd.Flags().GetString("image")
+
+		if len(image) == 0 {
+			return fmt.Errorf("image flag must be specified")
```

Comment:
- might be wise not to tie the create command to only creating drives from OCI images. the constants are the fstype, size and name (or path to put it). whether we want an  image burned into the loop drive should be optional

Reply:
The idea of create was to have a bootable VM at the end. And that is tied into the init creating since there are metadata in the pull phase that are shared with the init drive creation phase. These two aren't exactly decoupled. But we can explore the idea of actually writing the config files to disk and dynamically reading and creating out init if need be. Does that sound good to you?